include/vcpkg/base/cache.h

```diff
@@ -29,9 +29,9 @@ namespace vcpkg
     {
         template<class KeyIsh,
                  class F,
-                 typename std::enable_if<std::is_constructible<Key, const KeyIsh&>::value &&
-                                             detail::is_callable<Compare&, const Key&, const KeyIsh&>::value,
-                                         int>::type = 0>
+                 std::enable_if_t<std::is_constructible<Key, const KeyIsh&>::value &&
+                                      detail::is_callable<Compare&, const Key&, const KeyIsh&>::value,
```

Comment:
does this have a `_v` version ?

Reply:
We don't use variable templates right now and I don't want to affect our compiler support matrix. I took the _t change because we were already using enable_if_t elsewhere.

Reply:
> affect our compiler support matrix

is that public somewhere ?
according to cppreference it should be available for c++17 (libstdc++ 7 ; libc++ 3.8)

Reply:
> > affect our compiler support matrix
> 
> is that public somewhere ? according to cppreference it should be available for c++17 (libstdc++ 7 ; libc++ 3.8)

Currently, the only thing we "officially support" is the set of platforms we produce binaries for. But we also get people who are using "unsupported" configurations that we don't want to gratuitously break. For example https://github.com/microsoft/vcpkg-tool/pull/359 . As a result we try not to change the set of features we're using unless we have good reason to do so.

Reply:
>according to cppreference it should be available for c++17 (libstdc++ 7 ; libc++ 3.8)

libstdc++ or compiler version don't matter. What is in distros matters. For example the table in https://github.com/microsoft/vcpkg/discussions/19048